File: frontend/src/app/LSP/DiagramContentChangeEvent.ts
import { IShape } from '../entity/IShape';
import { Phenomenon } from '../entity/Phenomenon';
import { LineInfo } from '../entity/LineInfo';
import { Project } from '../entity/Project';
export interface DiagramContentChangeEvent{
    shapeType: string; //"mac/pro/req/int/ref/con//phe//project"
    changeType: string; //"add/delete/change"

    //if shapeType in "mac/pro/req/int/ref/con"
    oldShape?: IShape;
    newShape?: IShape;

    //if shapeType == "phe"
    lineInfo?: LineInfo;
    oldPhenomenon?: Phenomenon;
    newPhenomenon?: Phenomenon;

    newProject:Project;
}

File: backend/src/main/java/com/example/demo/LSP/bean/DiagramContentChangeEvent.java
package com.example.demo.LSP.bean;

import com.example.demo.bean.Phenomenon;
import com.example.demo.bean.Shape;

public class DiagramContentChangeEvent {
    String shapeType;
    String changeType;

    //if shapeType in "mac/pro/req/int/ref/con"
    Shape oldShape;
    Shape newShape;

    //if shapeType == "phe"
    LineInfo lineInfo;
    Phenomenon oldPhenomenon;
    Phenomenon newPhenomenon;
	public String getShapeType() {
		return shapeType;
	}
	public void setShapeType(String shapeType) {
		this.shapeType = shapeType;
	}
	public String getChangeType() {
		return changeType;
	}
	public void setChangeType(String changeType) {
		this.changeType = changeType;
	}
	public Shape getOldShape() {
		return oldShape;
	}
	public void setOldShape(Shape oldShape) {
		this.oldShape = oldShape;
	}
	public Shape getNewShape() {
		return newShape;
	}
	public void setNewShape(Shape newShape) {
		this.newShape = newShape;
	}
	public LineInfo getLineInfo() {
		return lineInfo;
	}
	public void setLineInfo(LineInfo lineInfo) {
		this.lineInfo = lineInfo;
	}
	public Phenomenon getOldPhenomenon() {
		return oldPhenomenon;
	}
	public void setOldPhenomenon(Phenomenon oldPhenomenon) {
		this.oldPhenomenon = oldPhenomenon;
	}
	public Phenomenon getNewPhenomenon() {
		return newPhenomenon;
	}
	public void setNewPhenomenon(Phenomenon newPhenomenon) {
		this.newPhenomenon = newPhenomenon;
	}
    
}
